### DISCUSSION TITLE
Handling Static Linking with Different Versions of Abseil in Libraries

### DISCUSSION THREAD

**Discussion by mudream4869** [abseil/abseil-cpp#1757] [General]
I noticed versioning in the symbol table of an LTS version (by `nm`). Does this mean that if two libraries are statically linked to different versions of Abseil and they don’t expose the Abseil API, our program can still statically link to both libraries without issues?

**Comment by derekmauro** [comment]
I believe you are asking about the inline namespaces. They are for handling this situation, though I'm not sure how well the mechanism has been tested in practice.

https://github.com/abseil/abseil-cpp/blob/f7c22f52a748761b93c50a27b06afef000b26f95/absl/base/options.h#L206-L229
